controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1611,6 +1616,88 @@ bool master_thread::handle_cmdu_1905_topology_notification(const std::string &sr
     return true;
 }
 
+bool master_thread::handle_cmdu_1905_topology_response(const std::string &src_mac,
+                                                       ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    auto mid = cmdu_rx.getMessageId();
+    LOG(DEBUG) << "Received TOPOLOGY_RESPONSE_MESSAGE, mid=" << std::hex << int(mid);
+
+    auto tlvDeviceInformation = cmdu_rx.getClass<ieee1905_1::tlvDeviceInformation>();
+    if (!tlvDeviceInformation) {
+        LOG(ERROR) << "ieee1905_1::tlvDeviceInformation not found";
+        return false;
+    }
+
+    const auto &al_mac = tlvDeviceInformation->mac();
+
+    // Get all Agent known fronthaul radios if exist, and compare it to what the Agent reports.
+    // If a radio exist on the database but not the tlvDeviceInformation, this radio node needs
+    // to be removed.
+    // This shall work in the opposite way as well: If it doesn't exist on the database, but
+    // reported in the TLV, then we need to add the node to the database. Of course this will remain
+    // as "TODO" for future task. Meanwhile parse properly the TLV, leaving the unused parts in
+    // comment for future implementation.
+    auto fronthaul_radios_on_db = database.get_node_children(
+        network_utils::mac_to_string(al_mac), beerocks::TYPE_SLAVE, beerocks::STATE_CONNECTED);
+
+    std::unordered_set<std::string> reported_fronthaul_radios;
```

Comment:
```suggestion
    std::unordered_set<sMacAddress> reported_fronthaul_radios;
```

Reply:
I did it because of the database return `set` of `std::string`.
I don't want to change the database in this PR scope.
Let's give #524 more priority.

Reply:
Still, the decision is to use `sMacAddress` instead of `std::string` even if we don't change everything at once, so please change to `sMacAddress` as part of this PR.

Reply:
Indeed. It's OK for `fronthaul_radios_on_db` because that is indeed returned by the database. But `reported_fronthaul_radios` gets filled with the MAC addresses that come from the device information so they're not strings. And where you do `reported_fronthaul_radios.find(fronthaul_radio_on_db)` you can easily do `reported_fronthaul_radios.find(network_utils::string_to_mac(fronthaul_radio_on_db))` instead.

Anyway, I'm not going to block the merge for this. But please take it into account next time.